lib/ansible/plugins/connection/network_cli.py

```diff
@@ -349,7 +350,18 @@ def receive(self, command=None, prompts=None, answer=None, newline=True, prompt_
         self._matched_cmd_prompt = None
         matched_prompt_window = window_count = 0
 
+        timeout = self._ssh_shell.gettimeout()
         while True:
+            start = time.time()
+            while True:
```

Comment:
Why is the reason to add second while loop here? Does the call to  `recv_ready()` increase the delay amount.

Will the below work?
```
while True:
   if self._ssh_shell.recv_ready():
     data = self._ssh_shell.recv(256)
     <--snip-->
```

Reply:
imo the waiting data timeout has to be managed and so a waiting loop is needed before calling self._ssh_shell.recv. I put that in a function and tried to rebase my commit for the changes you did in PR #47004 (i'm not sure i did that correctly, i did something wrong at first). Does it seem correct to you ?